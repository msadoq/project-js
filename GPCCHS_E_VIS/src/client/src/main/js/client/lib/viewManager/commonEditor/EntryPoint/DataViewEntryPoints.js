import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import EntryPointActionsContainer from 'viewManager/commonEditor/EntryPoint/EntryPointActionsContainer';
import EntryPointTreeContainer from 'viewManager/common/Components/Editor/EntryPointTreeContainer';
import ErrorBoundary from 'viewManager/common/Components/ErrorBoundary';
import EntryPointConnectedDataFieldsContainer
  from 'viewManager/common/Components/Editor/EntryPointConnectedDataFieldsContainer';


/**
 * Corresponding views:
 *  - TextView
 *  - MimicView
 */
export default class DataViewEntryPoints extends PureComponent {
  static propTypes = {
    viewId: PropTypes.string.isRequired,
    pageId: PropTypes.string.isRequired,
    search: PropTypes.string,
    viewType: PropTypes.oneOf(['TextView', 'MimicView', 'HistoryView', 'PlotView']).isRequired,
    // from container
    entryPoints: PropTypes.arrayOf(PropTypes.shape()),
    entryPointConnectedDataForm: PropTypes.func,
  };

  static defaultProps = {
    search: null,
    entryPoints: [],
    entryPointConnectedDataForm: EntryPointConnectedDataFieldsContainer,
  };

  render() {
    const {
      entryPoints,
      viewId,
      pageId,
      search,
      viewType,
      entryPointConnectedDataForm,
    } = this.props;
    return (
      <ErrorBoundary>
        <React.Fragment>
          <EntryPointActionsContainer
            viewId={viewId}
            search={search || undefined} // will use EntryPointActions' default value if null
            viewType={viewType}
          />
          <EntryPointTreeContainer
            viewId={viewId}
            pageId={pageId}
            entryPoints={entryPoints}
            search={search}
            entryPointConnectedDataForm={entryPointConnectedDataForm}
          />
        </React.Fragment>
      </ErrorBoundary>
    );
  }
}
