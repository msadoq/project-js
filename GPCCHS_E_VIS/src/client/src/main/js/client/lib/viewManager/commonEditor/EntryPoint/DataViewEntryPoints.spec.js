import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DataViewEntryPoints from './DataViewEntryPoints';

const propsStub = {
  domains: [
    {
      domainId: 1,
      itemNamespace: 'Domains',
      name: 'fr.cnes.isis',
      oid: '0051525005151000565215465660515',
      parentDomainId: 0,
    },
    {
      domainId: 2,
      itemNamespace: 'Domains',
      name: 'fr.cnes.isis.simupus',
      oid: '0051525005151000565215465660515',
      parentDomainId: 1,
    },
  ],
  domainName: 'fr.cnes.test',
  viewId: 'viewId',
  pageId: 'pageId',
  form: 'form',
  viewType: 'TextView',
};

describe('DataViewEntryPoints :: render', () => {
  test('DataViewEntryPoints :: render', () => {
    const tree = shallow(
      <DataViewEntryPoints
        {...propsStub}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
