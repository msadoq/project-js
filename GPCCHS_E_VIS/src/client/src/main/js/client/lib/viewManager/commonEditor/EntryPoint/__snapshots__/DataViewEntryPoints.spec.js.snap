// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewManager viewManager :: commonEditor viewManager :: commonEditor :: EntryPoint viewManager :: commonEditor :: EntryPoint :: DataViewEntryPoints snapshot 1`] = `
<Unknown>
  <Connect(EntryPointActions)
    search={undefined}
    viewId="viewId"
    viewType="TextView"
  />
  <Connect(EntryPointTree)
    entryPointConnectedDataForm={[Function]}
    entryPoints={Array []}
    pageId="pageId"
    search={null}
    viewId="viewId"
  />
</Unknown>
`;
